c++/src/kj/compat/tls.c++

```diff
@@ -443,8 +531,31 @@ public:
   }
 
 private:
+  Promise<void> doOneAccept() {
+    // Accept one connection and queue up the next accept on our TaskSet
+
+    return inner->acceptAuthenticated().then(
+        [this](AuthenticatedStream&& stream) {
+          auto promise = kj::evalNow([&] {
+            return tls.wrapServer(kj::fwd<kj::AuthenticatedStream>(stream));
+          });
+
+          tasks.add(queue.push(kj::mv(promise)));
+          tasks.add(doOneAccept());
+        },
+        [this](Exception&& e) {
+          KJ_LOG(ERROR, "TlsConnectionReceiver could not accept()", kj::fwd<Exception>(e));
+
+          // Even though we had a bad accept(), we should still loop again.
```

Comment:
Many errors thrown by `accept()` are permanent, meaning that if you call it again, it's just going to throw again, and you'll have a hot loop, which is pretty bad. Think of ENOTSOCK, for example.

Generally we have therefore taken the convention that an exception thrown by `accept()` must be presumed to be permanent. And as a result of that, `accept()` implementations need to be careful not to throw errors that they know to be one-offs. The latter rule is tricky and we've definitely screwed it up a few times, but it doesn't seem like there's any way around it.

Reply:
Hmmmm, I think my calculous on the hot loop looks like this: If we have to manually restart the process, then the hot loop with errors is actually better because we get _something_ done in that time frame. If we automatically restart the process, then the hot loop is obviously worse. I don't know what our situation is there.

Reply:
On the tricky throwing conventions...I think we can probably make that less painful if we separate the fatal error path into a separate fulfiller. Errors that flow through the loop would be retriable. Errors that flow through the termination process are most definitely not.

Reply:
> loop with errors is actually better because we get something done in that time frame.

Unfortunately, there a bunch of reasons that doesn't really work out:

* The KJ event loop likes to do depth-first scheduling because it's more efficient (better cache efficacy, etc.) and more predictable (fewer opportunities for a concurrent task to jump in and change things out from under you), but this means during a hot loop, no other events in the queue will make progress.
* For similar reasons, the KJ event loop also will not check for events from the OS until the in-memory queue is empty. So a hot loop will cause the thread to stop making any progress on I/O.
* Even if the KJ event loop tried to be a fair scheduler, a hot loop would tend to win the competition for CPU time over better-behaved tasks, leading better-behaved tasks to perform very badly. This is true both within the loop's thread itself, and across the system -- Linux is pretty bad at scheduling interactive tasks in the presence of CPU-saturating tasks.

So what would actually happen in this case is the process would saturate the CPU for 30-60 seconds, not make any progress on any I/O, and eventually get killed by the watchdog.

Reply:
Hmmmm, that's good context to have. Out of curiosity, is there a maximum "depth" of the scheduling? I'd imagine at some point we'd be able to deduce that we're starving out other async sources.